strawberryfields/backends/states.py

```diff
@@ -1110,6 +1148,30 @@ def poly_quad_expectation(self, A, d=None, k=0, phi=0, **kwargs):
 
         return mean, var
 
+    def number_expectation(self, modes):
+        """
+        Calculates the expectation value of the product of the number operators of the modes.
+        Args:
+            modes (list): list of modes for which one wants the expectation of the product of their number operator.
+        Return:
+            (float): the expectation value.
+        """
+        if len(modes) != len(set(modes)):
+            raise ValueError("there can be no duplicates in modes")
+        mu = self._mu
+        cov = self._cov
+        if len(modes) == 1:
+            return photon_number_mean(mu, cov, modes[0], hbar=self._hbar)
+
+        if len(modes) == 2:
+            ni = photon_number_mean(mu, cov, modes[0], hbar=self._hbar)
+            nj = photon_number_mean(mu, cov, modes[1], hbar=self._hbar)
+            return photon_number_covar(mu, cov, modes[1], modes[0], hbar=self._hbar) + ni * nj
+
+        raise ValueError("number_expectation only supports one or two modes")
```

Comment:
```suggestion
        raise ValueError("The number_expectation method only supports one or two modes for Gaussian states.")
```

Reply:
This should also be specified in the docstring, either using a `.. warning::` or in the `Args:` description.